metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,457 @@
+    type=str,
+    help="Change namespace from the default (your username) to the one specified.",
+)
+@click.argument("tags", required=True, type=str, nargs=-1)
+@click.pass_obj
+def remove(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = get_run_client(obj, run_id, user_namespace)
+
+    metaflow_run_client.remove_tags(tags)
+
+    obj.echo("Operation successful. New tags:")
+    print_tags(obj, metaflow_run_client)
+
+
+@tag.command("replace", help="Replace tags of a run.")
+@click.option(
+    "--run-id",
+    required=False,  # set False here so we can throw a better error message
+    default=None,
+    type=str,
+    help="Run ID of the specific run to tag. [required]",
+)
+@click.option(
+    "--namespace",
+    "user_namespace",
+    required=False,
+    default=None,
+    type=str,
+    help="Change namespace from the default (your username) to the one specified.",
+)
+@click.argument("tags", required=True, type=str, nargs=2)
+@click.pass_obj
+def replace(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = get_run_client(obj, run_id, user_namespace)
+
+    metaflow_run_client.replace_tag(tags[0], tags[1])
+
+    obj.echo("Operation successful. New tags:")
+    print_tags(obj, metaflow_run_client)
+
+
+@tag.command("list", help="List tags of a run.")
+@click.option(
+    "--run-id",
+    required=False,
+    default=None,
+    type=str,
+    help="Run ID of the specific run to list.",
+)
+@click.option(
+    "--all",
+    "list_all",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="List tags across all runs of this flow.",
+)
+@click.option(
+    "--my-runs",
+    "my_runs",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="List tags across all runs of the flow under the default namespace.",
+)
+@click.option(
+    "--hide-system-tags",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Hide system tags.",
+)
+@click.option(
+    "--by-tag",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Display results by showing runs grouped by tags",
+)
+@click.option(
+    "--by-run",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Display tags grouped by run",
+)
+@click.argument("arg_run_id", required=False, default=None, type=str)
+@click.pass_obj
+def tag_list(
+    obj, run_id, hide_system_tags, list_all, my_runs, by_tag, by_run, arg_run_id
+):
+    if run_id is None and arg_run_id is None and not list_all and not my_runs:
+        # Assume list_all by default
+        list_all = True
+
+    if list_all and my_runs:
+        raise CommandException(
+            "Option --all cannot be used together with --my-runs.\n"
+            "You can run the flow with commands 'tag list --help' "
+            "to check the meaning of the two options."
```

Comment:
Ideally we should indicate the meaning right here rather than expect the user to type another command to get the information needed.